### PR TITLE
[DOC-2733] Additional matrix examples

### PR DESCRIPTION
**Pull request by rashminandasahoo**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by naidu-harness — APPROVED** [review]

File: docs/platform/8_Pipelines/w_pipeline-steps-reference/additional-matrix-examples.md [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+
+For detailed information on using matrix in your pipeline, go to [Run a Stage or Step Multiple Times using a Matrix](https://developer.harness.io/docs/platform/Pipelines/run-a-stage-or-step-multiple-times-using-a-matrix)
+
+## Runtime input support in matrix
+
+### Support axis in matrix as the runtime input
+
+To define an axis as runtime input, mark it as `<+input>`.
+
+When you execute the pipeline, you are prompted to enter its value, which you can add as an array.
```
Comment (SudheendraKatte) [inline docs/platform/8_Pipelines/w_pipeline-steps-reference/additional-matrix-examples.md:17]: When you run the pipeline, you will be prompted to provide the matrix value. You can  add it as an array.

File: docs/platform/8_Pipelines/w_pipeline-steps-reference/additional-matrix-examples.md [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+
+Let us look at an example.
+
+If you wish to run your test on different versions of Go, the list of different versions will be available at runtime. You can set the axis of the matrix as an input at runtime in this case as : 
+
+```
+GoVersion: <+input>
+```
+
+Here is the YAML of a sample pipeline where the versions of Go are defined as runtime input:
```
Comment (SudheendraKatte) [inline docs/platform/8_Pipelines/w_pipeline-steps-reference/additional-matrix-examples.md:29]: The following YAML file shows a sample pipeline in which different versions of Go are defined as runtime inputs:

File: docs/platform/8_Pipelines/w_pipeline-steps-reference/additional-matrix-examples.md [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+                  strategy:
+                    GoVersion:
+                      - 1.1.0
+                      - 1.1.1
+
+```
+
+#### Support a few axes of matrix as runtime inputs
+
+When you have a matrix with more than one axis, you can set a few of the axes as runtime inputs.
```
Comment (SudheendraKatte) [inline docs/platform/8_Pipelines/w_pipeline-steps-reference/additional-matrix-examples.md:90]: If your matrix has more than one axis, you can set some of them as runtime input.

File: docs/platform/8_Pipelines/w_pipeline-steps-reference/additional-matrix-examples.md [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+  identifier: RepeatJiraTickets
+  variables:
+    - name: jiraTickets
+      type: String
+      value: cd-1,cd-2,cd-3 // We have added the list of tickets to be comma separated
+
+```
+## Provide the axis value of matrix from triggers
+
+To provide the axis value from a trigger, define the axis values which has a comma separated list of string.
```
Comment (SudheendraKatte) [inline docs/platform/8_Pipelines/w_pipeline-steps-reference/additional-matrix-examples.md:253]: To provide the axis value from a trigger, define the axis values, which have a comma separated list of strings.

**Review by SudheendraKatte — APPROVED** [review]
Approved with suggestions

File: docs/platform/8_Pipelines/w_pipeline-steps-reference/additional-matrix-examples.md (hunk 4 above)
Comment (rashminandasahoo) [reply docs/platform/8_Pipelines/w_pipeline-steps-reference/additional-matrix-examples.md:253]: There is a single list, so this should be "has a comma-separated list of strings."

**Review by SudheendraKatte — APPROVED** [review]

**Pull request merged**